### ISSUE TITLE
Mail history contains Mails with different topics

### ISSUE THREAD

**Issue by davidpr78** [nextcloud/mail#4509]
### Expected behavior
The mail history of one conversation/topic ist displayed

### Actual behavior
The mail history of one conversation contains mails with different topics (but same recipients/senders)

### Mail app

**Mail app version:**
1.8.1

**Mailserver or service:**
WebhostOne

### Server configuration

Edudocs.org

#### Client configuration
**Browser:** 
Chrome 88

**Operating system:** (e.g. Arch Linux)
Windows 10

**Comment by ChristophWurst** [comment]
any more info on the relation of the messages? do they have the same subject perhaps? senders do not matter for thread building

**Comment by davidpr78** [comment]
That's the point: They do not have the same subject. It appears when more than two people are included in the conversation.

**Comment by ChristophWurst** [comment]
Alright, here's what we do. [Enable debug mode](https://docs.nextcloud.com/server/stable/developer_manual/getting_started/devenv.html#enabling-debug-mode), then open that conversation. On the top right of each of the messages there is a ... menu where you can export the thread data.
![Bildschirmfoto von 2021-02-09 19-56-15](https://user-images.githubusercontent.com/1374172/107413317-ea54b200-6b10-11eb-9a62-ffffea021234.png)

Please download that data for all of the messages in the conversation and share it with us. If subjects or senders are sensible then redact/replace the values but make sure to replace them with a unique string.

Thanks.

**Comment by davidpr78** [comment]
ok, here we go:
{
  "subject": "Wocheplan KW6",
  "messageId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>",
  "inReplyTo": null,
  "references": [],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}
{
  "subject": "Wocheplan KW6",
  "messageId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>",
  "inReplyTo": null,
  "references": [],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}
{
  "subject": "Re: Aufgaben KW5",
  "messageId": "<CAD5DMq4g0MqCaOG9vHV=gm2H0W2hvHbd5RSSeJ+kV2TuHhANKA@***>",
  "inReplyTo": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>",
  "references": [
    "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
  ],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}
{
  "subject": "Re: Wocheplan KW5",
  "messageId": "<CAD5DMq5c4YV85pUhnJ2Hf_s_+zF-R38S0NG5iN3p0NuTttS8kQ@***>",
  "inReplyTo": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>",
  "references": [
    "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
  ],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}
{
  "subject": "Re: Wocheplan KW6",
  "messageId": "<15E87E85-0C9F-411F-962B-0E60E131063F@***>",
  "inReplyTo": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>",
  "references": [
    "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
  ],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***"
}
{
  "subject": "Videokonferenz ***",
  "messageId": "<trinity-e57fbd4e-8f18-4775-959f-9aa7bf1c7c79-1612784385957@***>",
  "inReplyTo": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>",
  "references": [
    "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
  ],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}
{
  "subject": "Märchen",
  "messageId": "<trinity-c1abd565-d3e9-4aac-b0f4-f121213b5ec2-1612786446172@***>",
  "inReplyTo": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>",
  "references": [
    "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
  ],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}
{
  "subject": "Re: ***",
  "messageId": "<20210208122602.Horde.W0QbmieHjszvP8KJt8O-bFd@***>",
  "inReplyTo": "<trinity-e57fbd4e-8f18-4775-959f-9aa7bf1c7c79-1612784385957@***>",
  "references": [
    "<trinity-e57fbd4e-8f18-4775-959f-9aa7bf1c7c79-1612784385957@***>"
  ],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}
{
  "subject": "Re: Märchen",
  "messageId": "<20210208123327.Horde.Zie4QbWBE5ks_jwhYlN8Ksr@***>",
  "inReplyTo": "<trinity-c1abd565-d3e9-4aac-b0f4-f121213b5ec2-1612786446172@***>",
  "references": [
    "<trinity-c1abd565-d3e9-4aac-b0f4-f121213b5ec2-1612786446172@***>"
  ],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}
{
  "subject": "Aw: Re: Märchen",
  "messageId": "<trinity-49305ea1-f052-4df6-a6c3-e8f96efaba56-1612789911699@***>",
  "inReplyTo": "<20210208123327.Horde.Zie4QbWBE5ks_jwhYlN8Ksr@***>",
  "references": [
    "<trinity-c1abd565-d3e9-4aac-b0f4-f121213b5ec2-1612786446172@***>",
    "<20210208123327.Horde.Zie4QbWBE5ks_jwhYlN8Ksr@***>"
  ],
  "threadRootId": "<20210207134213.Horde.pLWJnRNt87eeGIfPVUo79Xo@***>"
}

**Comment by ChristophWurst** [comment]
Thanks a lot!

**Comment by ChristophWurst** [comment]
Some debugging:

* `Wocheplan KW6` shows twice, assuming that is because two Nextcloud users got the same message
* `Re: Aufgaben KW5`, `Re: Wocheplan KW6`, `Märchen` and `Videokonferenz ***` are stored as children of `Wocheplan KW6`. This is correct because both inReplyTo and references point to it. -> I assume the user who sent those emails used the *Reply* option in their email client, but to send an unrelated email. The email client will send reference headers to the original message though.
* `Re: ***` is also part of the same thread because it is sent with inReplyTo/references pointing to `Videokonferenz ***`.
* `Re: Märchen` and `Aw: Re: Märchen` have inReplyTo/references point to `Märchen`

to sum up: the threads are calculated correctly. This is a user error. They use their email software wrong. The email software sends reference headers for unrelated messages.

Hope this makes sense :v:

**Comment by davidpr78** [comment]
Thanks for your check!!
